### DISCUSSION TITLE
Is there any clarity on when GA release will be for S3TransferManger ?

### DISCUSSION THREAD

**Discussion by fykidwai** [aws/aws-sdk-java-v2#3256] [General]
Is there any clarity on when GA release will be for S3TransferManger ?

**Comment by debora-ito** [comment]
I know it's frustrating but we don't comment about release schedules in general.

I can say that we are actively working on TransferManager v2 in preparation for the GA release. We want to include some features before GA (see comment [here](https://github.com/aws/aws-sdk-java-v2/issues/37#issuecomment-1162060981)) and it's also going through a standard process of security reviews.

Feedbacks are welcome, feel free to write us about your experience using the TransferManager v2.

**Comment by kingan379** [comment]
@debora-ito Do you know if  new S3TransferManager will have the method for aborting an upload?

**Reply by debora-ito** [reply]
Zoe answered you question here: https://github.com/aws/aws-sdk-java-v2/issues/37#issuecomment-1204184027

**Comment by kingan379** [comment]
@debora-ito Is there a plan to support InputStream in S3TransferManager? And is it known if it's gonna be released this year?

**Comment by yasminetalby** [comment]
Hello @kingan379 ,

Unfortunately, as @debora-ito mentioned above, we do not comment about the release schedule.
In general, if you have any questions or comments, all informations regarding the S3TransferManager for AWS Java SDK v2 will be provided in #37 .

As @zoewangg mentioned in the latest update , the features that will be included in the GA release of the AWS Java SDK v2 S3TransferManager are:

- download an S3 object to a file or any destination
- upload a single object of any content (if it's not a file, content-length must be supplied)
- download all objects in a bucket to a local directory
- upload all files in a directory recursively to an S3 bucket
- copy data from one Amazon S3 location to another Amazon S3 location.
pause an ongoing single file download and resume it at a later time
- pause an ongoing single file upload and resume it at a later time

We will use separate issues to track features that are not in GA scope (they will get prioritized based on the number of 👍🏼s).
You can submit any feature request [here](https://github.com/aws/aws-sdk-java-v2/issues/new/choose) as well as check on previous feature request status under the [issue tab of our repository.](https://github.com/aws/aws-sdk-java-v2/issues)

We would love for you to provide feedback about your experience using the TransferManager v2 [here](https://github.com/aws/aws-sdk-java-v2/issues/37#issuecomment-1204184027). It is very valuable to our team.

Thank you very much for your interest in the S3TransferManager for AWS Java SDK v2! 

Best regards,

Yasmine
